### ISSUE TITLE
CronSchedule workflow issue

### ISSUE THREAD

**Issue by zydovech** [temporalio/temporal#1818]
## Expected Behavior
     
    my CronSchedule is   */2 * * * *  。
    I expect to be executed every two minutes
    
## Actual Behavior
    
  Sometimes it was executed twice in two minutes

## Steps to Reproduce the Problem

  1. run cron workflow
  
   i see the change in https://github.com/temporalio/temporal/pull/560 , 

  I think the following changes caused multiple implementation issues
  
```
	if nowUTC.Before(scheduledUTCTime) { //this code is new for 560 。
		nextScheduleTime = scheduledUTCTime
	} else {
		nextScheduleTime = schedule.Next(scheduledUTCTime)
		// Calculate the next schedule start time which is nearest to now (right after now).
		for nextScheduleTime.Before(nowUTC) {
			nextScheduleTime = schedule.Next(nextScheduleTime)
		}
	}
```

## Specifications

  - Version:  v1.11.2
  - Platform: linux

**Comment by yiminc** [comment]
This PR https://github.com/temporalio/temporal/pull/1828 should have fixed this cron schedule issue.

**Comment by dnr** [comment]
After talking with @wxing1292 , we're not really sure that his PR fully fixes this issue (though it may help). There still might be another bug here, so we'll keep it open while I investigate more.

**Comment by zydovech** [comment]
I am not too clear about the whole process of workflow, I am still learning， I looked at the code of cadence，
There is no ```nowUTC.Before(scheduledUTCTime) ``` judgment . 
 
in my case ,  nowUTC is before scheduledUTCTime, nextScheduleTime is set to scheduledUTCTime，Resulted in another dispatch immediately，After I deleted the if judgment, directly set nextScheduleTime = schedule.Next(scheduledUTCTime), there is no repeated execution

**Comment by dnr** [comment]
Could you please try again with the just-released 1.11.4 patch release? It includes two fixes to cron scheduling that we believe should address this.

I've looked at the code you pointed out and I believe it's correct. In general, that clause shouldn't occur in normal operation: `now` should always be after `scheduled`, otherwise we wouldn't be checking the next backoff. The reason you're seeing it is because of separate bugs that caused `scheduled` to be set to an incorrect value too far in the future.

**Comment by zydovech** [comment]
Hi , i try again , this problem have fixed . thanks a lot 。
